File: 3. Stack/Stack Implementation With Linked List/LinkedStack.java

import java.util.Scanner;

public class LinkedStack {
    private int data;
    private LinkedStack next;
    private LinkedStack top = null;

    private LinkedStack createNode(int val) {
        try{
            LinkedStack node = new LinkedStack();
            node.data = val;
            node.next = null;
            return node;
        } catch(Exception e) {
            System.out.println("Failed");
        }
        return null;
    }

    // private void push() {
        // System.out.print("\nEnter Your Number:- ");
        // int val = myObj.nextInt();
    // struct node *newNode;
    // newNode = createNode(val);
    // newNode -> next = top;
    // top = newNode;
    // cout <<"Insertion Successfull. Value:- " << val << "\n";
    // }

    public static void main(String[] args) {
        Scanner myObj = new Scanner(System.in);
        LinkedStack stack = new LinkedStack();

        int choice;
        System.out.println("Hello! Here You Can Perform Following Stack Operations!");
        while (true) {
            System.out.println("1. Push(Insert)\n2. Pop(Delete)\n3. Peek(See Last Element)\n4. Is Empty(Checks weather Stack is Empty)\n5. Display All Elements\n6. Exit");
            System.out.print("Enter Your Choice:- ");
            choice = myObj.nextInt();

            switch (choice) {
                case 1:
                    System.out.print("\nEnter Your Number:- ");
                    int val = myObj.nextInt();
                    LinkedStack newNode = stack.createNode(val);
                    newNode.next = stack.top;
                    stack.top = newNode;
                    System.out.format("\nInsertion Successfull:- %d\n", val);
                    break;
                case 2:
                    if (stack.top == null) {
                        System.out.println("\n~~Stack is Empty~~\n");
                        break;
                    }
                    System.out.format("\nDeletion Successfull:- %d\n", stack.top.data);
                    stack.top = stack.top.next;
                    break;
                case 3:
                    if (stack.top == null) {
                        System.out.println("\n~~Stack is Empty~~\n");
                    } else {
                        System.out.format("\nData Element:- %d\n", stack.top.data);
                    }
                    break;
                case 4:
                    if (stack.top == null) {
                        System.out.println("\n~~Stack is Empty~~\n");
                    } else {
                        System.out.println("\n~~Stack is Not Empty~~\n");
                    }
                    break;
                case 5:
                    if (stack.top == null) {
                        System.out.println("\n~~Stack is Empty~~\n");
                        break;
                    }

                    LinkedStack current = stack.top;
                    System.out.print("\nLinked Stack: ");
                    while (current != null) {
                        System.out.format("%d ", current.data);
                        current = current.next;
                    }
                    System.out.print("\n\n");
                    break;
                case 6:
                    System.out.println("\n~~Thanks for Using! Have a Great Day!~~\n");
                    myObj.close();
                    return;
                default:
                    System.out.println("\n~~Please Enter a Valid Input!~~\n");
            }
        }
    }
}

File: 3. Stack/Stack Implementation With Linked List/linkedStack.py
top = None;

class linkedStack:
    def __init__(self, data, next):
        self.data = data;
        self.next = next;

def createNode(val):
    newNode = linkedStack(val, None);
    return newNode;

print("Hello! Here You Can Perform Following tack Operation!");
while (True) :
    print("1. Push(Insert)\n2. Pop(Delete)\n3. Peek(See Last Element)\n4. Is Empty(Checks weather Stack is Empty)\n5. Display All Elements\n6. Exit");
    choice = int(input("Enter Your Choice:- "));
    if(choice == 1):
        val = int(input("\nEnter Your Number:-"));
        newNode = createNode(val);
        newNode.next = top;
        top = newNode;
        print("Insertion Successfull:- ", val, "\n");
        continue;
    elif(choice == 2):
        if (top == None):
            print("\n~~ Stack is Empty ~~\n");
            continue;
        print("\nDeletion Successfull:- ", top.data, "\n");
        top = top.next;
        continue;
    elif(choice == 3):
        if (top == None):
            print("\n~~ Stack is Empty ~~\n");
        else: 
            print("\nData Element:- ", top.data, "\n");
        continue;
    elif(choice == 4):
        if (top == None):
            print("\n~~ Stack is Empty ~~\n");
        else: 
            print("\n~~ Stack is Not Empty ~~\n");
        continue;
    elif(choice == 5):
        if (top == None):
            print("\n~~ Stack is Empty ~~\n");
            continue;
        current = top;
        print("\nLinked Stack: ");
        while(current != None):
            print(current.data, " ");
            current = current.next;
        print("\n");
        continue;
    elif(choice == 6):
        print("\n~~ Thanks For Using! Have a Great Day! ~~\n");
        break;
    else:
        print("\n~~ Please Enter a Vaild Input! ~~\n");
        continue;